addons/survey/security/survey_security.xml

```diff
@@ -97,88 +79,87 @@
-            <field name="domain_force">[(1, '=', 1)]</field>
+            <field name="domain_force">[('survey_id.survey_type', 'in', ('assessment', 'custom', 'live_session', 'survey'))]</field>
             <field name="groups" eval="[(4, ref('group_survey_manager'))]"/>
             <field name="perm_unlink" eval="1"/>
             <field name="perm_write" eval="1"/>
             <field name="perm_read" eval="1"/>
             <field name="perm_create" eval="1"/>
         </record>
-        <record id="survey_user_input_rule_survey_user_read" model="ir.rule">
-            <field name="name">Survey user input: officer: read all</field>
-            <field name="model_id" ref="survey.model_survey_user_input"/>
-            <field name="domain_force">[(1, '=', 1)]</field>
-            <field name="groups" eval="[(4, ref('group_survey_user'))]"/>
-            <field name="perm_unlink" eval="0"/>
-            <field name="perm_write" eval="0"/>
-            <field name="perm_read" eval="1"/>
-            <field name="perm_create" eval="0"/>
-        </record>
-        <record id="survey_user_input_rule_survey_user_cw" model="ir.rule">
-            <field name="name">Survey user input: officer: create/write/unlink linked to own survey only</field>
-            <field name="model_id" ref="survey.model_survey_user_input"/>
-            <field name="domain_force">[('survey_id.create_uid', '=', user.id)]</field>
-            <field name="groups" eval="[(4, ref('group_survey_user'))]"/>
-            <field name="perm_unlink" eval="1"/>
-            <field name="perm_write" eval="1"/>
-            <field name="perm_read" eval="0"/>
-            <field name="perm_create" eval="1"/>
-        </record>
 
         <record id="survey_user_input_line_rule_survey_manager" model="ir.rule">
             <field name="name">Survey user input line: manager: all</field>
             <field name="model_id" ref="survey.model_survey_user_input_line"/>
-            <field name="domain_force">[(1, '=', 1)]</field>
+            <field name="domain_force">[('user_input_id.survey_id.survey_type', 'in', ('assessment', 'custom', 'live_session', 'survey'))]</field>
             <field name="groups" eval="[(4, ref('group_survey_manager'))]"/>
             <field name="perm_unlink" eval="1"/>
             <field name="perm_write" eval="1"/>
             <field name="perm_read" eval="1"/>
             <field name="perm_create" eval="1"/>
         </record>
-        <record id="survey_user_input_line_rule_survey_user_read" model="ir.rule">
-            <field name="name">Survey user input line: officer: read all</field>
-            <field name="model_id" ref="survey.model_survey_user_input_line"/>
-            <field name="domain_force">[(1, '=', 1)]</field>
+        
+        <!-- Keep seperate rules for Read and CWU, for compatibility with hr_recruitment_survey addon -->
+        <record id="survey_user_input_rule_survey_user_read" model="ir.rule">
```

Comment:
This is overriden in hr_recruitment_survey... So, once it is installed, this domain will be replaced with `('applicant_id', '=', False)`, which will absolutely ignore `survey_type` and `restrict_user_ids`